crates/stages-api/src/pipeline/mod.rs

```diff
@@ -234,22 +236,38 @@ where
     ///
     /// CAUTION: This method locks the static file producer Mutex, hence can block the thread if the
     /// lock is occupied.
-    pub fn produce_static_files(&self) -> RethResult<()> {
+    pub fn move_to_static_files(&self) -> RethResult<()> {
         let mut static_file_producer = self.static_file_producer.lock();
 
-        let provider = self.provider_factory.provider()?;
-        let targets = static_file_producer.get_static_file_targets(HighestStaticFiles {
-            headers: provider
-                .get_stage_checkpoint(StageId::Headers)?
-                .map(|checkpoint| checkpoint.block_number),
-            receipts: provider
-                .get_stage_checkpoint(StageId::Execution)?
-                .map(|checkpoint| checkpoint.block_number),
-            transactions: provider
-                .get_stage_checkpoint(StageId::Bodies)?
-                .map(|checkpoint| checkpoint.block_number),
-        })?;
-        static_file_producer.run(targets)?;
+        // Copies data from database to static files
+        let lowest_static_file_height = {
+            let provider = self.provider_factory.provider()?;
+            let stages_checkpoints = [StageId::Headers, StageId::Execution, StageId::Bodies]
+                .into_iter()
+                .map(|stage| {
+                    provider.get_stage_checkpoint(stage).map(|c| c.map(|c| c.block_number))
+                })
+                .collect::<Result<Vec<_>, _>>()?;
+
+            let targets = static_file_producer.get_static_file_targets(HighestStaticFiles {
+                headers: stages_checkpoints[0],
+                receipts: stages_checkpoints[1],
+                transactions: stages_checkpoints[2],
+            })?;
+            static_file_producer.run(targets)?;
+            stages_checkpoints.into_iter().min().expect("exists")
+        };
+
+        // Deletes data which has been copied to static files.
+        if let Some(prune_tip) = lowest_static_file_height {
+            // Run the pruner so we don't potentially end up with higher height in the database vs
+            // static files during a pipeline unwind
+            let mut pruner = PrunerBuilder::new(Default::default())
+                .prune_delete_limit(usize::MAX)
+                .build(self.provider_factory.clone());
```

Comment:
I think it also requires overriding the timeout
https://github.com/paradigmxyz/reth/blob/5b4e10cbdcb66dd4b6a20175c31f9b93659774f0/crates/prune/src/builder.rs#L105-L105

Reply:
added a large timeout, but ideally it would be None. 

we agreed on a follow-up PR changing the default timeout to be non existent instead, so this large timeout is only temporary